SecurityPkg/Library/Tpm2HelpLib/Tpm2Help.c

```diff
@@ -291,31 +320,93 @@ CopyDigestListToBuffer (
 }
 
 /**
-  Copy a buffer into a TPML_DIGEST_VALUES structure.
+  Get TPML_DIGEST_VALUES data size.
+
+  @param[in]     DigestList    TPML_DIGEST_VALUES data.
+
+  @return TPML_DIGEST_VALUES data size.
+**/
+UINT32
+EFIAPI
+GetDigestListSize (
+  IN TPML_DIGEST_VALUES  *DigestList
+  )
+{
+  UINTN   Index;
+  UINT16  DigestSize;
+  UINT32  TotalSize;
+
+  TotalSize = sizeof (DigestList->count);
+  for (Index = 0; Index < DigestList->count; Index++) {
+    DigestSize = GetHashSizeFromAlgo (DigestList->digests[Index].hashAlg);
+    TotalSize += sizeof (DigestList->digests[Index].hashAlg) + DigestSize;
+  }
+
+  return TotalSize;
+}
+
+/**
+  This function get digest from digest list.
+
+  @param[in]  HashAlg       Digest algorithm
+  @param[in]  DigestList    Digest list
+  @param[out] Digest        Digest
+
+  @retval EFI_SUCCESS       Digest is found and returned.
+  @retval EFI_NOT_FOUND     Digest is not found.
+**/
+EFI_STATUS
+EFIAPI
+GetDigestFromDigestList (
+  IN TPMI_ALG_HASH       HashAlg,
+  IN TPML_DIGEST_VALUES  *DigestList,
+  OUT VOID               *Digest
+  )
+{
+  UINTN   Index;
+  UINT16  DigestSize;
+
+  DigestSize = GetHashSizeFromAlgo (HashAlg);
+  for (Index = 0; Index < DigestList->count; Index++) {
+    if (DigestList->digests[Index].hashAlg == HashAlg) {
+      CopyMem (
+        Digest,
+        &DigestList->digests[Index].digest,
+        DigestSize
+        );
+      return EFI_SUCCESS;
+    }
+  }
+
+  return EFI_NOT_FOUND;
+}
+
+/**
+  Copy a buffer into  TPML_DIGEST_VALUES structure.

```

Comment:
Now that you are making this a standalone libraryclass linkable by multiple modules (and even implemented by different instances) you need to harden the code and define more.  Buffer format should be define and validated.  More parameter checking is also needed.